xchange-okcoin/src/main/java/org/knowm/xchange/okcoin/v3/service/OkexFuturesTradeService.java

```diff
@@ -0,0 +1,357 @@
+package org.knowm.xchange.okcoin.v3.service;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order.OrderType;
+import org.knowm.xchange.dto.marketdata.Trades.TradeSortType;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.OpenOrders;
+import org.knowm.xchange.dto.trade.UserTrade;
+import org.knowm.xchange.dto.trade.UserTrades;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.exceptions.NotAvailableFromExchangeException;
+import org.knowm.xchange.okcoin.OkexAdaptersV3;
+import org.knowm.xchange.okcoin.OkexExchangeV3;
+import org.knowm.xchange.okcoin.v3.dto.trade.OkexOpenOrder;
+import org.knowm.xchange.okcoin.v3.dto.trade.OkexOrderFlags;
+import org.knowm.xchange.okcoin.v3.dto.trade.OkexTradeHistoryParams;
+import org.knowm.xchange.okcoin.v3.dto.trade.OkexTransaction;
+import org.knowm.xchange.okcoin.v3.dto.trade.OrderCancellationRequest;
+import org.knowm.xchange.okcoin.v3.dto.trade.OrderCancellationResponse;
+import org.knowm.xchange.okcoin.v3.dto.trade.OrderPlacementResponse;
+import org.knowm.xchange.okcoin.v3.dto.trade.OrderPlacementType;
+import org.knowm.xchange.okcoin.v3.dto.trade.Side;
+import org.knowm.xchange.okcoin.v3.dto.trade.SpotOrderPlacementRequest;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.CancelOrderByCurrencyPair;
+import org.knowm.xchange.service.trade.params.CancelOrderByIdParams;
+import org.knowm.xchange.service.trade.params.CancelOrderParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.orders.DefaultOpenOrdersParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParams;
+
+public class OkexFuturesTradeService extends OkexTradeServiceRaw implements TradeService {
```

Comment:
now we will have two classes that implement TradeService  for an exchanges?

what will happen if someone will pass an order to an futures contract to `OkexTradeService#placeLimitOrder` ?

this makes no sense - the whole point of doing a generic Instrument class was to handle futures in the same implementation